api/handlers_buckets.go

```diff
@@ -87,6 +88,29 @@ func (s *server) BucketCreateHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	rollBackTasks = append(rollBackTasks, rbfunc)
 
+	// wait for the bucket to exist
+	err = retry(3, 2*time.Second, func() error {
+		log.Infof("checking if bucket exists before continuing: %s", bucketName)
+		exists, err := s3Service.BucketExists(r.Context(), bucketName)
+		if err != nil {
+			return err
+		}
+
+		if exists {
+			log.Infof("bucket %s exists", bucketName)
+			return nil
+		}
+
+		msg := fmt.Sprintf("s3 bucket (%s) doesn't exist", bucketName)
+		return errors.New(msg)
```

Comment:
Maybe I don't understand how this retry works, but isn't this going to return an error after the very first retry if the bucket doesn't exist?